src/Core/MySQL/MySQLReplication.cpp

```diff
@@ -584,6 +585,19 @@ namespace MySQLReplication
                         }
                         break;
                     }
+                    case MYSQL_TYPE_BIT:
+                    {
+                        UInt32 bits = ((meta >> 8) * 8) + (meta & 0xff);
+                        UInt32 size = (bits + 7) / 8;
+
+                        Bitmap bitmap_value;
+                        String byte_buffer;
+                        byte_buffer.resize(size);
+                        readBigEndianStrict(payload, reinterpret_cast<char *>(byte_buffer.data()), size);
+                        readBitmapFromStr(byte_buffer.c_str(), bitmap_value, size);
+                        row.push_back(Field{UInt64{bitmap_value.to_ulong()}});
```

Comment:
It it equivalent to simply casting `char *` to `UInt64 *`? I mean something like that:
```
UInt64 bitmap;
assert(size <= sizeof(bitmap));
readBigEndianStrict(payload, reinterpret_cast<char *>(&bitmap), size);
row.push_back(bitmap);
```

Reply:
yes, the two ways of this are the same, however your as shown above  it looks more elegantly and concise. i will improve it.

Reply:
done!